traits/src/storage.rs

```diff
@@ -600,7 +600,6 @@ pub mod traits {
     pub trait GroupState<const VERSION: u16>: Entity<VERSION> {}
     pub trait GroupEpochSecrets<const VERSION: u16>: Entity<VERSION> {}
     pub trait LeafNodeIndex<const VERSION: u16>: Entity<VERSION> {}
-    pub trait GroupUseRatchetTreeExtension<const VERSION: u16>: Entity<VERSION> {}
```

Comment:
This is a breaking change when it doesn't get into v0.6.

But it also seems to changes the processing. What is the change here and should it be part of this PR?

Reply:
Good question. I noticed that `use_ratchet_tree_extension` exists in the `CoreGroupConfig` (or something like that) and in the `MlsGroupJoinConfig`. So as part of the "inlining" of `CoreGroup` into `MlsGroup`, I removed that superfluous config flag. By removing it, the storage provider functions became unused, so I removed those as well. So I'm not entirely sure it was strictly in-scope for this PR, but it felt weird doing the extra work of inlining a superfluous config flag when removing it was quicker 🤷🏻.

Reply:
I assume as much. At least now everything is reading and writing the same thing.